File: ruby/task3/14mod3.rb
def is_prime(n)
    return false if n < 2
    (2..Math.sqrt(n).to_i).each do |i| # Цикл for от 2 до sqrt(n) 
      return false if n % i == 0  
    end
    return true 
  end
  
  # Основная программа
  puts "Введите количество чисел:"
  n = gets.chomp.to_i 
  count = 0

  (1..n).each do |i|
    print "#{i} Число: "
    num = gets.chomp.to_i
    count += 1 if is_prime(num) # Проверка условия, если да то прибавляем 1 к count
  end

  puts "#{count} Простых чисел в массиве"
  

File: kotlin/task3/14mod3.kt
fun isPrime(n: Int): Boolean { // Функция для проверки простого числа
    if (n < 2) return false
    for (i in 2..Math.sqrt(n.toDouble()).toInt()) {
        if (n % i == 0) return false
    }
    return true
}

fun main() {
    println("Введите количество чисел")
    val n = readLine()?.toInt() ?: 0 // Ввод с клавиатуры
    var count = 0
    for (i in 0 until n) {
        println("${i + 1} Число: ")
        val num = readLine()?.toInt() ?: 0 // Ввод с клавиатуры
        if (isPrime(num)) count++
    }
    println("$count Простых чисел в массиве")
}
